### DISCUSSION TITLE
batch correction without ref control workflow -request for link

### DISCUSSION THREAD

**Discussion by klai001** [ImmuneDynamics/Spectre#152] [Q&A]
Hello,
ive saw from the tutorial that the batch correction requires some ref control samples.
however, my samples are randomised across the batches, i dont have a specific sample which i evaluate
across all my batch runs.

here in this link, there's a mention of the workflow for batch integration without ref control. but im unable to find the page with the relevant tutorial.
https://wiki.centenary.org.au/display/SPECTRE/Discovery+workflow+with+batch+alignment+using+CytoNorm

Could you kindly share w me the link for it?
thanks

**Comment by josephangelolopez** [comment]
Also in the same boat! If there's a workflow for batch integration without a reference control, would be great to give it a try!

**Reply by annabellenicolasmith** [reply]
Same here! I wonder if it's a workflow they're still planning to upload?
It's an excellent package & adding the option to integrate without a reference sample would really broaden the usability of it - I'd love to see it happen!

**Reply by ghar1821** [reply]
I can confirm something is cooking in the kitchen. Stay tuned!

**Reply by annabellenicolasmith** [reply]
That's amazing to hear! Thank you so much :) looking forward to it!

**Reply by betgou** [reply]
any update on this @ghar1821 ?

Thanks!

**Reply by annabellenicolasmith** [reply]
I can't speak for the official Spectre team, but I found a way around this by using [cyCombine](https://doi.org/10.1038/s41467-022-29383-5) (from [biosurf](https://github.com/biosurf/cyCombine)) between the transformation and clustering steps in Spectre. I hope this helps!

**Reply by betgou** [reply]
Thanks! Just found cyCombine on another thread so its reassuring to hear you've managed it. Thank you!

**Reply by betgou** [reply]
Do you have an example R script incorporating the cell.dat file @annabellenicolasmith ?

**Reply by ghar1821** [reply]
Sorry for the radio silence. I recently changed lab and only now have the chance to hop back on Spectre. 

We've written up `run.cycombine` and `run.rpca` functions, but they are all in the v2-dev branch (https://github.com/ImmuneDynamics/Spectre/tree/v2-dev). The functions should integrate well to existing Spectre based script. Please have a go and let us know if there are issues or bugs. I'll try to find the time to write up a simple tutorial over this weekend or the next one. 

You can either install the v2-dev branch using `remotes::install_github("ImmuneDynamics/Spectre", ref="v2-dev")` or just source the respective files (https://github.com/ImmuneDynamics/Spectre/blob/v2-dev/R/run.cycombine.R) or (https://github.com/ImmuneDynamics/Spectre/blob/v2-dev/R/run.rpca.R) and run the functions.

**Reply by betgou** [reply]
Great, thank you so much! Will have a look at this tomorrow and report back. Really appreciate your time.
